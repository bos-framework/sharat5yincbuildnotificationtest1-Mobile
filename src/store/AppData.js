import { AsyncStorage } from 'react-native';

class AppData {

  getItemForKey = async (key, callback) => {

    try {
      const value = await AsyncStorage.getItem(key);
      if (value !== null) {
        // We have data!!
        callback(true, value);
      }
    } catch (error) {
      // Error retrieving data
      callback(false, error.message);
    }

  };

  setItemForKey = async (key, value, callback) => {

    try {
      await AsyncStorage.setItem(key, value, () => {
        callback(true);
      });
    } catch (error) {
      callback(false);
    }
  }
  
}

export default new AppData();