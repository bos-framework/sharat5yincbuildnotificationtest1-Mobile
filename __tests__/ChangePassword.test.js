import 'react-native';
import React from 'react';
import ChangePasswordContainer from '../src/Screens/Container/changePasswordContainer';
import ChangePassword from '../src/Screens/Presentational/changePassword';
import store from '../src/store/store';


// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
describe('Render changepassword component without crashing', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<ChangePasswordContainer store={store} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    // Pass data to state of component.

    it('componentWillMount: set state values', () => {
        const component = shallow(<ChangePasswordContainer store={store} />);
        component.setState({ currentPassword: 'password' })
        expect(component.state('currentPassword')).toEqual('password');
    });

    // Find the controls/elements/child component in the ChangepasswordContainer component.

    it('Find components/Elements', () => {

        const component = shallow(<ChangePassword data={{
            userId: '',
            currentPassword: '',
            newPassword: '',
            confirmNewPasword: '',
            errors: {
                currentPasswordEmpty: false,
                newPasswordError: false,
                confirmNewPaswordError: false,
                passwordNotMatches: false
            }
        }} />);
        expect(component.find('Button').length).toEqual(1);
    });

    // Button action/press test. ToDo with all button controls.

    it('Button tap test', () => {

        let handleSaveMock = jest.fn()

        let component = renderer.create(<ChangePassword onSaveBtnTapped={handleSaveMock}
            data={{
                userId: '',
                currentPassword: '',
                newPassword: '',
                confirmNewPasword: '',
                errors: {
                    currentPasswordEmpty: false,
                    newPasswordError: false,
                    confirmNewPaswordError: false,
                    passwordNotMatches: false
                }
            }} />).getInstance()

        component.props.onSaveBtnTapped();
        expect(handleSaveMock).toHaveBeenCalled()

    });
});