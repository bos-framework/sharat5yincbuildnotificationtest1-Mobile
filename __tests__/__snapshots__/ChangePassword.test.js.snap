// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render changepassword component without crashing renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "marginTop": 50,
          "maxWidth": 270,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "borderColor": "#bdbdbd",
            "borderRadius": 0,
            "borderWidth": 1,
            "marginTop": 10,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="default"
          onChangeText={[Function]}
          placeholder="Current password"
          placeholderTextColor="#919090"
          rejectResponderTermination={true}
          returnKeyType="next"
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "#dbdbdb",
              "borderRadius": 0,
              "color": "#545454",
              "fontSize": 16,
              "height": 44,
              "paddingBottom": 0,
              "paddingLeft": 10,
              "paddingTop": 2,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "borderColor": "#bdbdbd",
            "borderRadius": 0,
            "borderWidth": 1,
            "marginTop": 10,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="default"
          onChangeText={[Function]}
          placeholder="New password"
          placeholderTextColor="#919090"
          rejectResponderTermination={true}
          returnKeyType="next"
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "#dbdbdb",
              "borderRadius": 0,
              "color": "#545454",
              "fontSize": 16,
              "height": 44,
              "paddingBottom": 0,
              "paddingLeft": 10,
              "paddingTop": 2,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "borderColor": "#bdbdbd",
            "borderRadius": 0,
            "borderWidth": 1,
            "marginTop": 10,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="default"
          onChangeText={[Function]}
          placeholder="Confirm new password"
          placeholderTextColor="#919090"
          rejectResponderTermination={true}
          returnKeyType="done"
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "#dbdbdb",
              "borderRadius": 0,
              "color": "#545454",
              "fontSize": 16,
              "height": 44,
              "paddingBottom": 0,
              "paddingLeft": 10,
              "paddingTop": 2,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#424142",
            "borderRadius": 2,
            "height": 44,
            "justifyContent": "center",
            "marginTop": 25,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 17,
            }
          }
        >
          Save
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
